### PR TITLE
improve performance when clearing notifications

### PR DESCRIPTION
**Pull request by connyduck**

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/util/NotificationHelper.java [hunk 1]
```diff
@@ -465,13 +468,18 @@ public static void disablePullNotifications() {
 
     public static void clearNotificationsForActiveAccount(@NonNull Context context, @NonNull AccountManager accountManager) {
         AccountEntity account = accountManager.getActiveAccount();
-        if (account != null) {
-            account.setActiveNotifications("[]");
-            accountManager.saveAccount(account);
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/util/NotificationHelper.java]: Database access! On the main thread! And in 90% of cases it's redundant!

**Comment by connyduck** [comment]
This should make the perf problems slightly less worse @koyuawsmbrtn

File: app/src/main/java/com/keylesspalace/tusky/util/NotificationHelper.java (hunk 1 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/util/NotificationHelper.java]: Could it be caught by `android.os.StrictMode`? Could you try it out please? Event `StrictMode.enableDefaults()` should be enough.

Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/util/NotificationHelper.java]: THIS is what causing lags for me! I thought it's some debugging in my branch but it's *this*. Good job finding it!

**Comment by koyuawsmbrtn** [comment]
Nice! Sign me up.

**Comment by connyduck** [comment]
No its not caught by `StrictMode.enableDefaults()`. Probably because SQLite only writes it to a cache?

**Comment by charlag** [comment]
Weird, I rememeber Room catching it and there was a trick to allow it while debugging.

**Comment by charlag** [comment]
Yup, unconditionally for some reason.
https://github.com/tuskyapp/Tusky/blob/master/app/src/main/java/com/keylesspalace/tusky/TuskyApplication.java#L72

**Comment by connyduck** [comment]
We should get rid of this `allowMainThreadQueries()`. I guess there are more main thread queries that are not that easy to remove though.

**Pull request merged**